### PR TITLE
Update upgrade-guide.html for KIP-960 and KIP-968

### PR DESCRIPTION
**Pull request by aliehsaeedii**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -134,6 +134,24 @@ <h3 class="anchor-heading"><a id="streams_notable_changes" class="anchor-link"><
     </p>
 
     <h3><a id="streams_api_changes_370" href="#streams_api_changes_370">Streams API changes in 3.7.0</a></h3>
+    <p>
+        IQv2 supports <code>MultiVersionedKeyQuery</code> (introduced in <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-968%3A+Support+single-key_multi-timestamp+interactive+queries+%28IQv2%29+for+versioned+state+stores">KIP-968</a>)
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]: `IQv2 adds support for retrieving`

File: docs/streams/upgrade-guide.html [hunk 2]
```diff
@@ -134,6 +134,24 @@ <h3 class="anchor-heading"><a id="streams_notable_changes" class="anchor-link"><
...
+        that allows retrieving a set of records with the same specified key and different timestamps within a specified time range.
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]: `with the same specified` -> `for a given key` ?

`and a specified time range.` (Maybe highlight that it's basically a historic query?)

File: docs/streams/upgrade-guide.html [hunk 3]
```diff
@@ -134,6 +134,24 @@ <h3 class="anchor-heading"><a id="streams_notable_changes" class="anchor-link"><
...
+        Users have to use <code>fromTime(Instant)</code> and/or <code>toTime(Instant)</code> to specify a half or a complete time range.
+        In the case of specifying no starting point and ending point for the time range, the query will return all the existing records associated with the specified key.
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]: Think we can omit this sentence? Too much detail?

We might want to update the "main IQ" docs though, additionally? IIRC, IQv2 is very sparsely (or maybe even not at all) explained.

File: docs/streams/upgrade-guide.html [hunk 4]
```diff
@@ -134,6 +134,24 @@ <h3 class="anchor-heading"><a id="streams_notable_changes" class="anchor-link"><
...
+        No ordering is guaranteed for the results, but the results can be sorted by timestamp in ascending or descending order by calling the corresponding defined methods
+        <code>withAscendingTimestamps()</code> or <code>withDescendingTimestamps()</code>, respectively.
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]: Similar?

File: docs/streams/upgrade-guide.html (hunk 1 above)
Comment (mjsax) [inline docs/streams/upgrade-guide.html]: The paragraph should mentioned that this new query type is for versioned state-stores, and thus add IQ support for the new versioned stores.

**Review by mjsax — COMMENTED** [review]
Thanks. A few suggestions.

Leave it up to you. Also ok to keep more details as this PR does currently.

**Comment by mjsax** [comment]
PR need to be rebased.

**Comment by aliehsaeedii** [comment]
> Thanks. A few suggestions.
> 
> Leave it up to you. Also ok to keep more details as this PR does currently.

Thanks. I followed your suggestions ;-)

**Pull request merged**